themes/pom.xml

```diff
@@ -11,6 +11,115 @@
     <artifactId>keycloak-themes</artifactId>
     <name>Keycloak Themes</name>
     <description/>
+    <packaging>jar</packaging>
+
+    <properties>
+        <dir.common>src/main/resources/theme/keycloak/common/resources</dir.common>
+        <dir.account2>src/main/resources/theme/keycloak-preview/account/resources</dir.account2>
+        <args.npm.install>install -P -E --no-optional --ignore-scripts --no-shrinkwrap</args.npm.install>
```

Comment:
The `--no-shrinkwrap` option causes that our package lock is ignored and regular package.json is used instead. That means we e.g. install newer dependencies than we expect (because we use `^` when specifying versions in package.json). And because of the newer dependencies the Account Console is broken. I know this is partially an issue with our package.json (we probably shouldn't use `^`) but at the same time I believe it's not a good practice to effectively ignore package lock.

Or did I miss something, do we need `--no-shrinkwrap` for some reason?

Reply:
It was part of the options that Stian added. I thought we were going to specify exact versions in package.json? If `npm install` is called every time maven runs, we don't want npm introducing noise into the commit with it randomly changing package-lock.json all the time.

Reply:
@ASzc I see. But shouldn't package lock also help us with transitive dependencies versions?

What about using [`npm ci`](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable) instead of `npm install`? It seems it's exactly what we need. It uses package lock to install dependencies, doesn't update it and produce "predictable" builds.

Reply:
The main disadvantage of `npm ci` is that it deletes the node_modules directory each time it's called, so some external caching would be needed to avoid a complete slowdown. I think npm does do this, in `~/.npm`, so I can swap the command over to `ci`